api/src/pcapi/core/users/external/user_automations.py

```diff
@@ -0,0 +1,166 @@
+    return add_contacts_to_list(
+        user_emails, settings.SENDINBLUE_AUTOMATION_YOUNG_18_IN_1_MONTH_LIST_ID, clear_list_first=True
+    )
+
+
+def get_users_beneficiary_credit_expiration_within_next_3_months() -> List[User]:
+    return (
+        User.query.yield_per(YIELD_COUNT_PER_DB_QUERY)
+        .options(load_only(User.email))
+        .join(User.deposits)
+        .filter(User.is_beneficiary.is_(True))
+        .filter(
+            Deposit.expirationDate.between(
+                datetime.combine(date.today(), datetime.min.time()),
+                datetime.combine(date.today() + relativedelta(days=90), datetime.max.time()),
+            )
+        )
+        .all()
+    )
+
+
+def users_beneficiary_credit_expiration_within_next_3_months_automation() -> bool:
+    """
+    This automation is called every day and includes all young user whose pass expires in the next 3 months. They may
+    have remaining unspent credit or not.
+
+    User enters in the list on the 90th day before expiration date and leaves the list the day after expiration date.
+
+    List: jeunes-expiration-M-3
+    """
+    user_emails = (user.email for user in get_users_beneficiary_credit_expiration_within_next_3_months())
+    return add_contacts_to_list(
+        user_emails, settings.SENDINBLUE_AUTOMATION_YOUNG_EXPIRATION_M3_ID, clear_list_first=True
+    )
+
+
+def get_users_ex_beneficiary() -> List[User]:
+    return (
+        User.query.yield_per(YIELD_COUNT_PER_DB_QUERY)
+        .options(load_only(User.email))
+        .join(User.deposits)
+        .filter(User.is_beneficiary.is_(True))
+        .filter(
+            or_(
+                Deposit.expirationDate <= datetime.combine(date.today(), datetime.min.time()),
+                and_(
+                    Deposit.expirationDate > datetime.combine(date.today(), datetime.min.time()),
+                    func.get_wallet_balance(User.id, False) <= 0,
+                ),
+            )
+        )
+        .all()
+    )
+
+
+def user_ex_beneficiary_automation() -> bool:
+    """
+    This automation is called every day to include all young users whose credit is expired: they either spent their full
+    credit or reached expiration date with unused credit.
+
+    Note that a young user may be in "jeunes-ex-benefs" list before "jeunes-expiration-M-3".
+
+    List: jeunes-ex-benefs
+    """
+    user_emails = (user.email for user in get_users_ex_beneficiary())
+    return add_contacts_to_list(
+        user_emails, settings.SENDINBLUE_AUTOMATION_YOUNG_EX_BENEFICIARY_ID, clear_list_first=False
+    )
+
+
+def get_inactive_user_since_thirty_days() -> List[User]:
+    date_30_days_ago = date.today() - relativedelta(days=30)
+
+    return get_young_users_emails_query().filter(func.date(User.lastConnectionDate) == date_30_days_ago).all()
+
+
+def users_inactive_since_30_days_automation() -> bool:
+    """
+    This automation called every day updates the list of users who are inactive since 30 days or more:
```

Comment:
> since 30 days **or more**

Pourtant, la requête plus haut filtre avec un `==`, pas un `<=`. Est-ce voulu ? D'après le commentaire plus bas (*even if the email has been sent marketing filters will ensure that he or she receives the email only once*), j'ai l'impression qu'on peut renvoyer plusieurs jours de suite le même utilisateur. Donc que la docstring ici est correcte, mais que la requête SQL plus haut devrait plutôt utiliser `<=` : `filter(func.date(User.lastConnectionDate) <= date_30_days_ago)`